test/unit/org/apache/cassandra/schema/LegacySchemaMigratorThriftTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.schema;
+
+import java.io.IOException;
+
+import org.junit.Test;
+
+import org.apache.cassandra.config.Schema;
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.exceptions.InvalidRequestException;
+import static org.apache.cassandra.cql3.QueryProcessor.executeOnceInternal;
+
+@SuppressWarnings("deprecation")
+public class LegacySchemaMigratorThriftTest extends LegacySchemaMigratorBaseTest
```

Comment:
@hanke580 btw what is "Thirft" about this? What was the problem that we could not have one class?

Reply:
**Thrift** means the legacy table is in thrift model created with cassandra-cli.
The reason they cannot be put into a single class is that between two tests, it need to clean the *data&commit log* folder. Otherwise there will be conflicts when creating/loading legacy data.

Reply:
@hanke580 why dont we clean data and commit log folder then? :)

Reply:
@hanke580 leave it is it is now ... I ve check that I do not have a clue why it throws even we clean it etc ... If separation into two classes helps, well ... that's just fine.

Reply:
Thanks! I also rechecked it. The reason is somehow related to a concurrent file descriptor read, but I am not sure.
I'll follow the previous methods.